domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs

```diff
@@ -531,4 +568,36 @@ public async Task CreateWalletDepositEndpoint_ShouldReturnUnauthorized_WhenUnaut
 
         result.StatusCode.Should().Be(HttpStatusCode.Unauthorized);
     }
+
+    [Fact]
+    public async Task Create_ShouldHaveReceiverReferenceNotNull()
+    {
+        var receiverId = Guid.NewGuid();
+
+        var result = await factory
+            .CreateAuthenticatedClient(sub: receiverId.ToString())
+            .PostAsync("api/transfer-agreements/wallet-deposit-endpoint", null);
+
+        var resultData = await result.Content.ReadFromJsonAsync<Dictionary<string, string>>();
+        var base64String = resultData!["result"];
+
+        var transferAgreement = new CreateTransferAgreement(
+            new DateTimeOffset(2123, 3, 3, 3, 3, 3, TimeSpan.Zero).ToUnixTimeSeconds(),
+            new DateTimeOffset(2124, 4, 4, 4, 4, 4, TimeSpan.Zero).ToUnixTimeSeconds(),
+            "12345456",
+            base64String
+        );
+
+        var response = await authenticatedClient.PostAsJsonAsync("api/transfer-agreements", transferAgreement);
+        response.EnsureSuccessStatusCode();
+
+        var createdTransferAgreement = await response.Content.ReadFromJsonAsync<TransferAgreementDto>();
+
+        await Task.Delay(2000);
+
+        var createdAgreementInDb = await context.TransferAgreements.FindAsync(createdTransferAgreement.Id);
```

Comment:
It can't, since it only returns the transfer agreement in dto format. the dto does not include the ReferenceGuid. That variable is stored in the database, behind the API layer. We need access to the database to access the specific field and its value